### PR TITLE
UsingPathDefinition: added positive and negative test cases

### PR DESCRIPTION
**Pull request by ppaulweber**
- related to casm-lang/casm#105
- related to casm-lang/libcasm-fe#195

### CODE REVIEW

**Comment by ppaulweber** [comment]
@emmanuel099 pls ignore `windows-x86_64` related build failures, GitHub action-based CI is still WIP

**Comment by ppaulweber** [comment]
@emmanuel099 pls review again, thx!

File: test/definition/using/path/make_all_enumeration_enumerator_symbols_visible.casm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//  may extend this exception to your version of the library, but you are
+//  not obliged to do so. If you do not wish to do so, delete this exception
+//  statement from your version.
+//
+
+CASM init test
+
+enumeration Color = { Red, Blue, Green }
+
+using Color::*
```
Comment (emmanuel099) [inline test/definition/using/path/make_all_enumeration_enumerator_symbols_visible.casm]: This should not be allowed IMHO.
May result in highly confusing code.

File: test/definition/using/path/make_explicit_enumeration_enumerator_symbol_visible.casm [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//  may extend this exception to your version of the library, but you are
+//  not obliged to do so. If you do not wish to do so, delete this exception
+//  statement from your version.
+//
+
+CASM init test
+
+enumeration Color = { Red, Blue, Green }
+
+using Color::Red
```
Comment (emmanuel099) [inline test/definition/using/path/make_explicit_enumeration_enumerator_symbol_visible.casm]: This should not be allowed IMHO.
May result in highly confusing code.

File: test/definition/using/path/make_all_enumeration_enumerator_symbols_visible.casm (hunk 1 above)
Comment (emmanuel099) [reply test/definition/using/path/make_all_enumeration_enumerator_symbols_visible.casm]: Take this as suggestion.

File: test/definition/using/path/make_explicit_enumeration_enumerator_symbol_visible.casm (hunk 2 above)
Comment (emmanuel099) [reply test/definition/using/path/make_explicit_enumeration_enumerator_symbol_visible.casm]: Take this as suggestion.

File: test/definition/using/path/make_all_enumeration_enumerator_symbols_visible.casm (hunk 1 above)
Comment (ppaulweber) [reply test/definition/using/path/make_all_enumeration_enumerator_symbols_visible.casm]: Then I decide that we are not allowing this in general and I'll add proper checks in the FE.

File: test/definition/using/path/make_explicit_enumeration_enumerator_symbol_visible.casm (hunk 2 above)
Comment (ppaulweber) [reply test/definition/using/path/make_explicit_enumeration_enumerator_symbol_visible.casm]: same as https://github.com/casm-lang/libcasm-tc/pull/84#discussion_r345773840

**Pull request merged**